src/arrayMethodJoin.js

```diff
@@ -3,9 +3,33 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let spliter = separator;
+
+    if (spliter !== {}) {
+      spliter += '';
+    }
```

Comment:
separator can't be an object, I don't think you need this if here

Reply:
![image](https://user-images.githubusercontent.com/36368111/134161682-dc202bd0-da79-4ac8-9662-8f756a3cb0bb.png)
But tests are falling without this check.